### ISSUE TITLE
Запись объектов - кодогенерация

### ISSUE THREAD

**Issue by aliczin** [1c-syntax/mdclasses#158]
просили issue - создадим

суть в следующем - сейчас библиотеки представляет модель для чтения - то есть сериализации из файлов в объект.
а хочется запись - то есть изменять объекты MD и десериализовать в файлы.

помимо всяких Ынтырпрайз сценариев использования есть один важный открытый сценарий

в Github встроили https://github.com/features/codespaces
да и не только туда - многие развертывают VSCode как отдельный сервис для Bitbucket/Gitlab, чтобы позволить кодировать прямо в браузере 

![изображение](https://user-images.githubusercontent.com/56105699/102752260-579e5f00-437a-11eb-8253-f5059a519225.png)

Такому сервису понадобится не только запись BSL языка, но еще и изменение метаданных.

То есть "справа" будет дерево метаданных с командами.
![изображение](https://user-images.githubusercontent.com/56105699/102752409-99c7a080-437a-11eb-99eb-735dc183e369.png)

для указанных сценариев и придется делать "запись", помимо чтения.
У меня сейчас в работе пилотный сценарий "Из DWH на Oracle в 1С метаданные" - но это уже другая история, хотя функциональность будет таже.

**Comment by asosnoviy** [comment]
@otymko переведи плз этот поток сознания в осмысленную таску =)

**Comment by aliczin** [comment]
@asosnoviy конечно-конечно - больше осмыслений. Больше ищузов и больше коментов. ;-). 


А существует какой-то формат по которой ты/Вы определяете осмысленность НЕосмысленность таски ? Лех забей - будет PR - сразу станет все понятно ;-)

**Comment by otymko** [comment]
@aliczin кейсы мне понятны. Что-то подобное для VSC предполагалось делать. Но описанная задача выше - она глобальная и даже трудно прикинуть трудозатраты на нее. Я бы начал со следующего:
* Выбор минимального кейса, например отрисовка дерева метаданных "крупными" мазками.
* Возможность в дереве метаданных выше менять порядок, с отражением изменений в формате edt / original.
* Сразу вырисовывается вопрос по взаимодействию VSC / Monako / Codespaces. Т.к. они на другом стеке (ts и т.п.), нужно будет организовывать между ними обмен. Через JSON RPC например.

Пример более сложный сходу на стороне текущего проекта не получится сделать - т.к. многих вещей не хватает в моделях метаданных.

**Comment by nixel2007** [comment]
я вообще не уверен, что взаимодействие с пользователем имеет смысл делать на стороне mdclasses. само апи сериализации - да, вполне. а отприсовка и UI/UX - это явно ide-specific задача, и для панели нужно делать отдельный плагин/доработку существующих плагинов. mdclasses все же довольно низкоуровневая библиотека.

эндпоинты для таких операций можно реализовать в виде protocol extension для bsl ls, дабы не утяжелять библиотеку. а panel api - уже задача плагина

**Comment by aliczin** [comment]
@nixel2007 именно API - в MDClasses именно API

**Comment by johnnyshut** [comment]
**Текущий MVP**  
Сейчас в writer’е используются только те поля, которые уже есть в модели (name, synonym, codeSeries, checkUnique и т.п.). Всё остальное пишется дефолтами в конвертерах.

**Вариант: JAXB + XSD**  
Можно писать XML через классы, сгенерированные по XSD-схемам платформы 1С (JAXB и т.п.). Тогда вывод будет на 100% совместим с выбранной версией платформы. Схемы берут из ресурсов платформы или выгружают через описание/XDTO в 1С.

```bsl
XMLReader = New XMLReader;
XMLReader.SetString(
  "<Model xmlns=""http://v8.1c.ru/8.1/xdto"" xmlns:xs=""http://www.w3.org/2001/XMLSchema"" xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xsi:type=""Model"">
  |  <package targetNamespace=""http://v8.1c.ru/8.3/MDClasses"" elementFormQualified=""true"" attributeFormQualified=""false"">
  |  </package>
  |</Model>"
);
XDTOModel = XDTOFactory.ReadXML(XMLReader);
MyXDTOFactory = New XDTOFactory(XDTOModel);
```
---
Фиксирую, чтобы не потерять.

**Comment by johnnyshut** [comment]
Частично, требования из текущего issues закрыл в [новом проекте](https://github.com/yellow-hammer/md-sparrow). 
Считаю, что в рамках `md-classes` изначальное требование не реализовать, иначе нужно много чего начать читать.
